File: entities/admin_action.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type AdminAction struct {
	ID             uuid.UUID `gorm:"primaryKey" json:"id"`
	AdminID        uuid.UUID `gorm:"not null;index" json:"admin_id"`
	DoctorID       uuid.UUID `gorm:"not null;index" json:"doctor_id"`
	Action         string    `gorm:"size:50;not null" json:"action"`
	PreviousStatus string    `gorm:"size:50;not null" json:"previous_status"`
	NewStatus      string    `gorm:"size:50;not null" json:"new_status"`
	Reason         *string   `gorm:"type:text" json:"reason,omitempty"`
	IPAddress      *string   `gorm:"size:50" json:"ip_address,omitempty"`
	UserAgent      *string   `gorm:"type:text" json:"user_agent,omitempty"`
	CreatedAt      time.Time `gorm:"autoCreateTime" json:"created_at"`
}

func (a *AdminAction) BeforeCreate(tx *gorm.DB) (err error) {
	if a.ID == uuid.Nil {
		a.ID = uuid.New()
	}
	return nil
}

File: python_backend/entities/admin_action.py
import uuid
from sqlalchemy import Column, String, DateTime, Text, ForeignKey
from sqlalchemy.sql import func
from database import Base

class AdminAction(Base):
    __tablename__ = "admin_actions"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    admin_id = Column(String(36), nullable=False, index=True)
    doctor_id = Column(String(36), ForeignKey("doctors.id", ondelete="CASCADE"), nullable=False, index=True)
    action = Column(String(50), nullable=False)
    previous_status = Column(String(50), nullable=False)
    new_status = Column(String(50), nullable=False)
    reason = Column(Text, nullable=True)
    ip_address = Column(String(50), nullable=True)
    user_agent = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
